bottlerocket-settings-models/modeled-types/src/shared.rs

```diff
@@ -945,6 +945,90 @@ string_impls_for!(Lockdown, "Lockdown");
 
 // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+/// ApiCommand represents a valid Bottlerocket API Command. It stores the command as a vector of
+/// strings and ensures that the first argument is apiclient.
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Default, Serialize, Deserialize)]
+pub struct ApiCommand(pub Vec<String>);
+
+impl TryFrom<Vec<String>> for ApiCommand {
+    type Error = error::Error;
+
+    fn try_from(input: Vec<String>) -> Result<Self, error::Error> {
+        let first_word = input.first().map(String::as_str);
+        ensure!(
+            matches!(first_word, Some("apiclient")),
+            error::InvalidApiCommandSnafu { input }
+        );
+
+        Ok(ApiCommand(input))
+    }
+}
+
+#[cfg(test)]
+mod test_valid_api_command {
+    use super::ApiCommand;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn valid_api_command() {
+        assert!(
+            ApiCommand::try_from(vec!["apiclient".to_string(), "motd=helloworld".to_string()])
+                .is_ok()
+        );
+    }
+
+    #[test]
+    fn empty_api_command() {
+        assert!(ApiCommand::try_from(Vec::new()).is_err());
+    }
+
+    #[test]
+    fn invalid_api_command() {
+        assert!(ApiCommand::try_from(vec![
+            "invalid".to_string(),
+            "arg1".to_string(),
+            "arg2".to_string()
+        ])
+        .is_err());
+    }
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// BootstrapCommandMode determines when the specified set of commands should be executed during
+/// the boot process of a system: 'always' runs them on every boot, 'once' runs them only on the
+/// first boot, and 'off' disables their execution.
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize, Default, Scalar)]
+#[serde(rename_all = "lowercase")]
+pub enum BootstrapCommandMode {
+    #[default]
+    Off,
+    Once,
+    Always,
+}
```

Comment:
I don't think we have other enums in these types. I'm not confident that this will serialize and deserialize properly.

You could think about renaming `BootstrapContainerMode` to `BootstrapMode` and reusing that type for both bootstrap containers and commands.

Reply:
We do use enums this way [elsewhere](https://github.com/bottlerocket-os/bottlerocket-settings-sdk/blob/291f07d214372809b29570db0589784432e2c4bc/bottlerocket-settings-models/modeled-types/src/ecs.rs#L177). The `scalar-derive` macro handles implementing the rest of the string-like traits.

Reply:
I have tested it with Unit tests and end to end testing of bootstrap commands feature and it works.

Reply:
We can continue having this discussion but for now I have repurposed BootstrapContainerMode by renaming it to BootstrapMode.